CRM/Contribute/Page/UserDashboard.php

```diff
@@ -83,11 +83,9 @@ public function listContribution() {
     $recurRow = array();
     $recurIDs = array();
     while ($recur->fetch()) {
-      $mode = $recur->is_test ? 'test' : 'live';
-      $paymentProcessor = CRM_Contribute_BAO_ContributionRecur::getPaymentProcessor($recur->id,
-        $mode
-      );
-      if (!$paymentProcessor) {
+      if (empty($recur->payment_processor_id)) {
```

Comment:
I hit an enotice in the tests & when I looked here I found that we have just done a dao->fetch so we already have the payment_processor_id - looks to me like the check is for something obsolete anyway - see orphan paymentObject later on... - probably this was to get the url but that's not in the function now